### PR TITLE
Bring back "Name-png" target

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

**Comment by rdumusc** [comment]
please rebase this commit! I see regressions in Daniel's past commit (d6cdb98)

**Comment by rdumusc** [comment]
other than that, +1

File: CommonTarget.cmake [hunk 1]
```diff
@@ -0,0 +1,11 @@
+
+# Creates an empty target, excluded it from the default build, and adds it to
```
Comment (tribal-tec) [inline CommonTarget.cmake]: ...target, exclude...

File: CommonGraph.cmake [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# Provides functions to generate dependency graph images using graphviz.
+# Used by common_package.
+# common_graph_dep(): Write a dependency from->to into global properties
+# common_graph(): Write .dot from the global properties and add Name-png rule
+
```
Comment (tribal-tec) [inline CommonGraph.cmake]: add a include guard?

File: CommonGraph.cmake [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    set(dest ${PROJECT_BINARY_DIR}/doc/images)
+    add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${Name}_tred.dot
+      COMMAND ${TRED_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/${Name}.dot >
+               ${CMAKE_CURRENT_BINARY_DIR}/${Name}_tred.dot
+      DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/${Name}.dot)
+    add_custom_command(OUTPUT ${dest}/${Name}.png
+      COMMAND ${CMAKE_COMMAND} -E make_directory ${dest}
+      COMMAND ${DOT_EXECUTABLE} -o ${dest}/${Name}.png -Tpng ${CMAKE_CURRENT_BINARY_DIR}/${Name}_tred.dot
+      DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/${Name}_tred.dot)
+    add_custom_target(${Name}-png DEPENDS ${dest}/${Name}.png)
```
Comment (tribal-tec) [inline CommonGraph.cmake]: target name Name-graph?

**Comment by eile** [comment]
all done.

**Pull request merged**